data-prepper-api/src/test/java/com/amazon/dataprepper/model/processor/AbstractProcessorTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  The OpenSearch Contributors require contributions made to
+ *  this file be licensed under the Apache-2.0 license or a
+ *  compatible open source license.
+ *
+ *  Modifications Copyright OpenSearch Contributors. See
+ *  GitHub history for details.
+ */
+
+package com.amazon.dataprepper.model.processor;
+
+import com.amazon.dataprepper.metrics.MetricNames;
+import com.amazon.dataprepper.metrics.MetricsTestUtil;
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.model.record.Record;
+import io.micrometer.core.instrument.Measurement;
+import io.micrometer.core.instrument.Statistic;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+
+class AbstractProcessorTest {
+
+    @Test
+    public void testMetrics() {
+        final String processorName = "testProcessor";
+        final String pipelineName = "pipelineName";
```

Comment:
Can we have the name of the pipeline different from the variable name? We can call it `testPipeline` or have a Random String. I've run into a false passing of unit tests with this before so want to recommend this :)